hermes/bin/src/wasm/patcher.rs

```diff
@@ -722,6 +735,94 @@ mod tests {
+                }
+            }
+        }
+
+        // Step 1: Patch the WASM file
+        let patcher =
+            Patcher::from_file(COMPONENT_SINGLE_CORE_MODULE).expect("should create patcher");
+        let result = patcher.patch().expect("should patch");
+        let encoded = wat::parse_str(&result).expect("should encode");
+
+        // Step 2: Instantiate the patched WASM
+        let engine = Engine::default();
+        let component =
+            wasmtime::component::Component::new(&engine, encoded).expect("should create component");
+        let mut linker = Linker::new(&engine);
+        add_to_linker_sync(&mut linker).expect("should add to linker");
+        let wasi = WasiCtxBuilder::new().inherit_stdio().inherit_env().build();
+        let ctx = MyCtx {
+            table: ResourceTable::new(),
+            wasi,
+        };
+        let mut store = Store::new(&engine, ctx);
+        let instance = linker
+            .instantiate(&mut store, &component)
+            .expect("should instantiate");
+
+        // Step 3: Call the init function which will put some bytes into the linear memory.
+        let init = instance
+            .get_func(&mut store, "init")
+            .expect("should get func")
+            .typed::<(), (bool,)>(&store)
+            .expect("should be a typed func");
+        let init_result = init.call(&mut store, ()).expect("should call").0;
+        init.post_return(&mut store).expect("should post return");
+        assert!(init_result);
+
+        // Step 4: Retrieve the linear memory size
+        let get_memory_size_func = format!("{MAGIC}get-memory-size");
+        let get_memory_size = instance
+            .get_func(&mut store, get_memory_size_func)
+            .expect("should get func")
+            .typed::<(), (i32,)>(&store)
+            .expect("should be a typed func");
+        let memory_size_in_pages = get_memory_size.call(&mut store, ()).expect("should call").0;
+        get_memory_size
+            .post_return(&mut store)
+            .expect("should post return");
+        let memory_size_in_bytes = memory_size_in_pages * LINEAR_MEMORY_PAGE_SIZE_BYTES;
+
+        // Step 5: Read the entire memory content
+        let mut linear_memory = vec![];
+        let get_memory_raw_bytes_func = format!("{MAGIC}get-memory-raw-bytes");
+        let get_memory_raw_bytes = instance
+            .get_func(&mut store, get_memory_raw_bytes_func)
+            .expect("should get func")
+            .typed::<(i32,), (i64,)>(&store)
+            .expect("should be a typed func");
+        for offset in (0..memory_size_in_bytes).step_by(8) {
+            let raw_bytes = get_memory_raw_bytes
+                .call(&mut store, (offset,))
+                .expect("should call")
+                .0;
+            get_memory_raw_bytes
+                .post_return(&mut store)
+                .expect("should post return");
+            linear_memory.extend(raw_bytes.to_le_bytes());
```

Comment:
Had to [look up](https://www.w3.org/TR/2019/REC-wasm-core-1-20191205/#control-instructions) WebAssembly integers. I didn't know it guarantees little endian byte order. Could be worth a brief mention?
```suggestion
            // > All values are read and written in little endian] byte order.
            // <https://www.w3.org/TR/2019/REC-wasm-core-1-20191205/#control-instructions>.
            linear_memory.extend(raw_bytes.to_le_bytes());
```